src/devices/Hcsr501/Hcsr501.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+
+namespace Iot.Device.Hcsr501
+{
+    public class Hcsr501ValueChangedEventArgs : EventArgs
+    {
+        public readonly PinValue PinValue;
+        public Hcsr501ValueChangedEventArgs(PinValue value)
+        {
+            PinValue = value;
+        }
+    }
+
+    public class Hcsr501 : IDisposable
+    {
+        private GpioController _controller;
+        private readonly int _pinOut;
+        private readonly PinNumberingScheme _pinNumberingScheme;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="pin">OUT Pin</param>
+        /// <param name="pinNumberingScheme">Pin Numbering Scheme</param>
+        public Hcsr501(int pin, PinNumberingScheme pinNumberingScheme)
+        {
+            _pinOut = pin;
+            _pinNumberingScheme = pinNumberingScheme;
+        }
+
+        /// <summary>
+        /// Initialize the sensor
+        /// </summary>
+        public void Initialize()
+        {
```

Comment:
see comments from other PRs - if you merge this with the constructor then _pinNumberingScheme doesn't need to be stored